src/components/date/__internal__/utils.js

```diff
@@ -103,7 +103,7 @@ function findMatchWithNoSeparators(valueString, formatString) {
   const valueArray = makeSeparatedValues(indexArray, valueString);
 
   if (checkForCompleteMatch(formatArray, valueArray)) {
-    return [formatArray.join("."), valueArray.join(".")];
+    return [formatArray.join(" "), valueArray.join(" ")];
```

Comment:
this was changed to prevent clashes with locales that use `.` as a separator, as the guard added to `onBlur` would not work for them